File: Techniques/Prefix Sum/findPivotIndex.py
class Solution:
    def pivotIndex(self, arr: list[int]) -> int:
        ans, l, r = -1, 0, sum(arr)

        for i, v in enumerate(arr):
            r -= v
            if l == r:
                ans = i
                break
            l += v
        return ans
        

File: Techniques/Prefix Sum/findPivotIndex.go
package main

func pivotIndex(arr []int) int {
	r := 0

	for _, v := range arr {
		r += v
	}

	ans, l := -1, 0

	for i, v := range arr {
		r -= v
		if l == r {
			ans = i
			break
		}
		l += v
	}
	return ans
}
